odoo/addons/base/models/res_company.py

```diff
@@ -251,7 +251,8 @@ def create(self, vals_list):
         return companies
 
     def write(self, values):
-        self.clear_caches()
+        if not self.env.context.get('skip_clear_ormcache'):
```

Comment:
Using context is often bad because it gets propagated often too far.
A good way to fix this would be to list fields that don't clear the cache.

Reply:
I see, I added a new method to get the fields that don't need to clear cache. Please review it.